Robust.Shared/Audio/Systems/SharedAudioSystem.cs

```diff
@@ -271,6 +274,36 @@ private void OnAudioGetStateAttempt(EntityUid uid, AudioComponent component, ref
         }
     }
 
+    private static void OnAudioPlayerFilterGetStateAttempt(
+        EntityUid uid,
+        AudioPlayerFilterComponent component,
+        ref ComponentGetStateAttemptEvent args)
+    {
+        // Only allow replay recording to capture this component.
+        // (or anybody if state attempts can be bypassed if PVS is disabled)
+        args.Cancelled |= args.Player == null;
```

Comment:
```suggestion
        args.Cancelled |= args.Player != null;
```

The component also needs to override the `Component.SessionSpecific` bool, otherwise the event never gets raised.

And this is probably somewhat beyond the scope of this PR, but if replay-only comp states become more common, we should probably add a separate check in `PvsSystem.GetEntityState` that doesn't bother raise an event and just checks if `player==null`.  And that should probably be done by replacing the current bools with enums or bitflags set via an attribute & cached in `MetaData.NetComponents` instead of overriding virtual bool getters in the base component class.